src/code_80005FD0.c

```diff
@@ -2510,64 +2510,64 @@ s16 func_8000CD24(f32 posX, f32 posY, f32 posZ, s16 waypointIndex, Player *playe
             }
             if (playerId == ((s32) D_80163488 % 8)) {
                 func_802ADDC8(&player->unk_110, 10.0f, posX, posY, posZ);
-                D_80163318[playerId] = func_802ABD40(player->unk_110.unk3A);
-                sp5E = func_8000C884(posX, posY, posZ, waypointIndex, pathIndex, D_80163318[playerId]);
-                if (sp5E == -1) {
-                    sp5E = func_8000C9DC(posX, posY, posZ, waypointIndex, pathIndex);
+                D_80163318[playerId] = get_track_segment(player->unk_110.collisionMeshIndex);
+                newWaypoint = func_8000C884(posX, posY, posZ, waypointIndex, pathIndex, D_80163318[playerId]);
+                if (newWaypoint == -1) {
+                    newWaypoint = find_closest_waypoint_with_previous_waypoint(posX, posY, posZ, waypointIndex, pathIndex);
```

Comment:
`find_next_waypoint`?

Reply:
there is multiple function who are similar so I choose a most detailled one